cookbook/asset/custom_version_strategy.rst

```diff
@@ -0,0 +1,205 @@
+                class: AppBundle\Asset\VersionStrategy\BusterVersionStrategy
+                arguments:
+                    - manifest_path: "%kernel.root_dir%/../busters.json"
+                      format: "%%s?version=%%s"
+                public: false
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+
+                <service id="app.assets.buster_version_strategy" class="AppBundle\Asset\VersionStrategy\BusterVersionStrategy" public="false">
+                    <argument type="collection">
+                        <argument key="manifest_path">%kernel.root_dir%/../busters.json</argument>
+                        <argument key="format">%%s?version=%%s</argument>
+                    </argument>
+                </service>
+
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $definition = new Definition(
+            'AppBundle\Asset\VersionStrategy\BusterVersionStrategy',
+            array(
+                array(
+                    'manifest_path' => '%kernel.root_dir%/../busters.json',
+                    'format' => '%%s?version=%%s',
+                ),
+            )
+        );
+        $definition->setPublic(false);
+
+        $container->setDefinition('app.assets.buster_version_strategy', $definition);
+
+Implement VersionStrategyInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    // src/AppBundle/Asset/VersionStrategy/BusterVersionStrategy.php
+    namespace AppBundle\Asset\VersionStrategy;
+
+    use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+    class BusterVersionStrategy implements VersionStrategyInterface
+    {
+        /**
+         * @var string
+         */
+        private $manifestPath;
+
+        /**
+         * @var string
+         */
+        private $format;
+
+        /**
+         * @var string[]
+         */
+        private $hashes;
+
+        /**
+         * @param array $options {
+         *     @var string      $manifest_path
+         *     @var string|null $format
+         * }
+         */
+        public function __construct(array $options)
+        {
+            $this->manifestPath = $options['manifest_path'];
+            $this->format = isset($options['format']) ? $options['format'] : '%s?%s';
+        }
+
+        public function getVersion($path)
+        {
+            if (!is_array($this->hashes)) {
+                $this->hashes = $this->loadManifest(array(
+                    'file_path' => realpath($this->manifestPath),
```

Comment:
why can't we do this in the ```private function loadManifest()```
```'file_path' => realpath($this->manifestPath),````

the method then could use the member var directly